### ISSUE TITLE
Issue with Arabic training

### ISSUE THREAD

**Issue by BSen007** [espnet/espnet#3954]
Hello all,

Recently I am trying to carry out some experiment (Epsnet1) using ~1000hrs of Arabic data. I have adapted from MGB2 Arabic recipe for this experiments. I have tried out several configs but did not succeed to get good results on held-out test sets. Here is one config that I have tried to transformer AM. I have two 16GB gps and based on my understand I have changed batch-bins and acc-grad, but I am not sure these two values are responsible for this or not.  Here is the Transformer config file and the learning curve is also attached (I can see validation is oscillating and training accuracy was below 0.4, even after 100 epochs),  
![acc_2](https://user-images.githubusercontent.com/93305829/148894495-2baa773d-6b11-4561-b276-e28c916ebac7.png)

```
# This configuration requires 4 gpus with 12GB memory
accum-grad: 8
adim: 256
aheads: 4
backend: pytorch
# https://github.com/espnet/espnet/issues/1167
batch-bins: 11000000
dlayers: 4
dropout-rate: 0.1
dunits: 1024
elayers: 10
epochs: 100
eunits: 1024
grad-clip: 5
lsm-weight: 0.1
model-module: espnet.nets.pytorch_backend.e2e_asr_transformer:E2E
mtlalpha: 0.3
opt: noam
patience: 0
sortagrad: 0
transformer-attn-dropout-rate: 0.0
transformer-init: pytorch
transformer-input-layer: conv2d
transformer-length-normalized-loss: false
transformer-lr: 5.0
transformer-warmup-steps: 20000
```

Can you please suggest me what should I do to overcome the issue?

**Comment by BSen007** [comment]
Any suggestion guys? Thanks

**Comment by sw005320** [comment]
Sorry for the late response.
I feel that the learning curve is too slow.
Can you test it by reducing the accum-grad etc.?
(I see that you try to keep the accum-grad x batch-bins to be the same as the original config in https://github.com/espnet/espnet/blob/master/egs/mgb2/asr1/conf/tuning/train_pytorch_transformer_large_ngpu4.yaml. So, you seem to do correctly, but the learning curve does not look good and it's worth tuning it).

@AmirHussein96, do you have a comment about it?

**Comment by BSen007** [comment]
Thanks for your suggestion! I will try with value 4. I guess, I need to increase batch-bins in this process, but I wonder if this cause OOM issue considering I do have 16GB GPUS.

**Comment by AmirHussein96** [comment]
Hi @BSen007, I use V100 with 16 GB with the original configuration https://github.com/espnet/espnet/blob/master/egs/mgb2/asr1/conf/tuning/train_pytorch_transformer_large_ngpu4.yaml. Are you changing the configuration because it causes OOM? In general I would expect that reducing the batch-bins by half will result in worse performance (2-5)% WER, but not as bad as in your case. So my suggestion is similar to @sw005320 , reduce the accum-grad to 4 and let us know if it works. In addition, try to increase the batch-bins as much as you can until you get OOM.

**Comment by BSen007** [comment]
Thanks, at present I am using accum-grad=4,  learning rate=2 and same batch-bins. 50 epoch/100 have been done, and the curve looks stable but training curve is below 0.4. So, I am thinking to relaunch the same with learning rate 5. Will update you.

**Comment by AmirHussein96** [comment]
Double-check your train dev sets, maybe there is something wrong with your data preparation. Double-check the number of hours of the training data after preparation.
